greenpithumb/sleep_windows.py

```diff
@@ -0,0 +1,84 @@
+"""Parses sleep window strings."""
+
+import datetime
+import re
+
+
+class Error(Exception):
+    pass
+
+
+class InvalidWindowFormatError(Error):
+    pass
+
+
+class HypersomniaError(Error):
+    pass
+
+
+def parse(sleep_windows):
+    """Converts sleep windows from strings to datetime objects.
+
+    Args:
+        sleep_windows: A list of sleep window strings.Each sleep window should
+            be in 24-hour HH:MM-HH:MM format.
+
+    Returns:
+        A list of 2-tuples, each representing a sleep window. Tuple items
+        are datetime.time objects.
+    """
+    awake_minutes = [1] * 60 * 24
+    sleep_minutes = set()
+    sleep_windows_datetime = []
+    for sleep_window in sleep_windows:
+        try:
+            window_match = re.match(r'(\d\d):(\d\d)-(\d\d):(\d\d)',
+                                    sleep_window)
+        except TypeError:
+            raise InvalidWindowFormatError(
+                'Time windows must be in string format.')
+        if not window_match:
+            raise InvalidWindowFormatError(
+                'Sleep windows must be in 24-hour HH:MM-HH:MM format')
+        try:
+            sleep_time = datetime.time(
+                int(window_match.group(1)), int(window_match.group(2)))
+            wake_time = datetime.time(
+                int(window_match.group(3)), int(window_match.group(4)))
+        except (ValueError, TypeError):
```

Comment:
Coveralls says we're never hitting this. I think because our invalid hour/minute cases are *too* off so the regex filters them. Can we add tests where it's like `'23:15-24:00'` and `'22:56-22:60'`?